MapleServer2/Servers/Game/DungeonManager.cs

```diff
@@ -77,37 +77,40 @@ public DungeonSession GetDungeonSessionByInstanceId(int instanceId)
         public bool IsDungeonUsingFieldInstance(DungeonSession dungeonSession, FieldManager fieldManager, Player player) //alternatively this could be: IsFieldInstanceUsed in FieldManagerFactory
         {
             //fieldmanager is the map the player is coming from/that is to be released
-            if (dungeonSession != null) //is not null after entering dungeon via directory
+            if (dungeonSession == null) //is not null after entering dungeon via directory
             {
-                //check map that is left: fieldmanager.mapId is the to be released map
-                if (dungeonSession.ContainsMap(fieldManager.MapId))  //map is a dungeon map
+                return false; //no dungeonsession = the map is unused by dungeon, 
+            }
+            //fieldManager.MapId: left map that is to be destroyed
+            //player.MapId: travel destination of the player
+            //check map that is left: 
+            if (dungeonSession.IsDungeonMap(fieldManager.MapId))  //left map is a dungeon map
```

Comment:
Likewise if you invert this and return false, it'll clear up this logic further

Reply:
oki!

Reply:
done, indeed much clearer.